packages/legend-extension-dsl-service/src/components/studio/ServiceRegisterModal.tsx

```diff
@@ -0,0 +1,376 @@
+      setOwners(options);
+    };
+    const onChangeServicePath: React.ChangeEventHandler<HTMLInputElement> = (
+      event,
+    ) => {
+      setServicePath(event.target.value);
+      setIsValidServicePath(isValidFullPath(event.target.value));
+    };
+
+    const onChangeServicePattern: React.ChangeEventHandler<HTMLInputElement> = (
+      event,
+    ) => {
+      setServicePattern(event.target.value);
+      setIsServiceUrlPatternValid(!validate_ServicePattern(event.target.value));
+    };
+    const serverRegistrationOptions =
+      editorStore.applicationStore.config.options
+        .TEMPORARY__serviceRegistrationConfig;
+
+    const envOptions = serverRegistrationOptions
+      .filter((options) =>
+        options.modes.includes(ServiceExecutionMode.SEMI_INTERACTIVE),
+      )
+      .map((info) => ({
+        label: info.env.toUpperCase(),
+        value: info.env,
+      }));
+    const selectedEnvOption = serviceEnv
+      ? {
+          label: serviceEnv.toUpperCase(),
+          value: serviceEnv,
+        }
+      : null;
+    const onServerEnvChange = (
+      val: { label: string; value: string } | null,
+    ): void => {
+      setServiceEnv(val?.value);
+    };
+    const toggleActivateService = (): void =>
+      setActivateService(!activateService);
+
+    const registerService = editorStore.applicationStore.guardUnhandledError(
+      async (): Promise<void> => {
+        const project = ProjectData.serialization.fromJson(
+          await editorStore.depotServerClient.getProject(
+            editorStore.query.groupId,
+            editorStore.query.artifactId,
+          ),
+        );
+        const currentQueryInfo =
+          await editorStore.graphManagerState.graphManager.getQueryInfo(
+            editorStore.query.id,
+          );
+        if (
+          registrationState.isInProgress ||
+          !servicePath ||
+          !servicePattern ||
+          !isServicePathValid ||
+          !isServiceUrlPatternValid ||
+          !selectedEnvOption
+        ) {
+          return;
+        }
+        try {
+          registrationState.inProgress();
+          const serverUrl = guaranteeNonNullable(
+            serverRegistrationOptions.find(
+              (option) => option.env === selectedEnvOption.value,
+            )?.managementUrl,
+          );
+          const versionInput = LATEST_PROJECT_REVISION;
+          registrationState.setMessage(`Registering service...`);
+          const service = await createServiceElement(
+            servicePath,
+            servicePattern,
+            owners,
+            currentQueryInfo.content,
+            currentQueryInfo.mapping,
+            currentQueryInfo.runtime,
+            editorStore.graphManagerState,
+          );
+          const serviceRegistrationResult =
+            await editorStore.graphManagerState.graphManager.registerService(
+              service,
+              editorStore.graphManagerState.graph,
+              project.groupId,
```

Comment:
for groupId,artifactId, versionId. Why are we fetching the project again. 
The query already holds everything we need.
`editorStore.query.groupId, editorStore.query.artifactId, editorStore.query.versionId`
If the version is head then the input is undefined for the registerService function